l10n_es_vat_prorate/models/account_move.py

```diff
@@ -3,116 +3,176 @@
+                )
+
+                non_deductible_balance = base_balance - deductible_balance
+                non_deductible_currency = base_currency - deductible_currency
+
+                line.balance = deductible_balance
+                line.amount_currency = deductible_currency
+
+                if not currency.is_zero(non_deductible_balance):
+                    expense_account = move._get_prorate_expense_account()
+
+                    if prorate_line_index < len(existing_prorate_lines):
+                        prorate_line = existing_prorate_lines[prorate_line_index]
+                        prorate_line.write(
+                            {
+                                "account_id": expense_account.id,
+                                "balance": non_deductible_balance,
+                                "amount_currency": non_deductible_currency,
+                                "currency_id": line.currency_id.id
+                                if line.currency_id
+                                else None,
+                                "partner_id": move.partner_id.id,
+                            }
+                        )
+                        prorate_line_index += 1
+                    else:
+                        new_line = move.env["account.move.line"].create(
+                            {
+                                "move_id": move.id,
+                                "account_id": expense_account.id,
+                                "balance": non_deductible_balance,
+                                "amount_currency": non_deductible_currency,
+                                "currency_id": line.currency_id.id
+                                if line.currency_id
+                                else None,
+                                "partner_id": move.partner_id.id,
+                                "vat_prorate": True,
+                                "display_type": "tax",
+                            }
+                        )
+                        existing_prorate_lines |= new_line
+                        prorate_line_index += 1
+                elif prorate_line_index < len(existing_prorate_lines):
+                    prorate_line = existing_prorate_lines[prorate_line_index]
+                    prorate_line.write(
+                        {
+                            "balance": 0.0,
+                            "amount_currency": 0.0,
+                        }
+                    )
+                    prorate_line_index += 1
+
+            for line in existing_prorate_lines[prorate_line_index:]:
+                line.write(
+                    {
+                        "balance": 0.0,
+                        "amount_currency": 0.0,
+                    }
+                )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        moves = super().create(vals_list)
+        moves._apply_vat_prorate()
+        return moves
+
+    def write(self, vals):
```

Comment:
Be careful with canceling the call to the super.
On the other hand, be careful with the application of the prorated rate, as there will be cases that are not covered by this method.